src/main/java/net/cardosi/mojo/cache/FileService.java

```diff
@@ -0,0 +1,136 @@
+package net.cardosi.mojo.cache;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+import io.methvin.watcher.DirectoryWatcher;
+import io.methvin.watcher.changeset.ChangeSet;
+import io.methvin.watcher.changeset.ChangeSetEntry;
+import io.methvin.watcher.changeset.ChangeSetListener;
+import io.methvin.watcher.hashing.FileHash;
+;
+
+public class FileService {
+
+    private final DirectoryWatcher          watcher;
+    private final ChangeSetListener         listener;
```

Comment:
ah so this is probably why the weird whitespace above. will find a formatter config to be consistent throughout the proj... in another commit.